node/src/components/contract_runtime/operations.rs

```diff
@@ -152,20 +155,21 @@ pub fn execute_finalized_block(
 
         trace!(?deploy_hash, ?result, "deploy execution result");
         // As for now a given state is expected to exist.
-        let (state_hash, execution_result) = commit_execution_results(
+        let (state_hash, execution_result, messages) = commit_execution_results(
             &scratch_state,
             metrics.clone(),
             state_root_hash,
             deploy_hash,
             result,
         )?;
-        execution_results.push((deploy_hash, deploy_header, execution_result));
+        execution_results.push((deploy_hash, deploy_header, execution_result, messages));
         state_root_hash = state_hash;
     }
 
     // Write the deploy approvals' and execution results' checksums to global state.
-    let execution_results_checksum =
-        compute_execution_results_checksum(execution_results.iter().map(|(_, _, result)| result))?;
+    let execution_results_checksum = compute_execution_results_checksum(
+        execution_results.iter().map(|(_, _, result, _)| result),
```

Comment:
Then you could hide the complexity

```suggestion
        execution_results.into_iter().map(|execution_result| execution_result.result),
```

Reply:
Changed in https://github.com/casper-network/casper-node/commit/53483e16590b67de9be9be4d9ba4e6b64d1f22ee